### PR TITLE
Add block support to UIControl

### PR DESCRIPTION
**Pull request by KingOfBrian**
This PR adds block support to UIControl using a simple associated object container.

### CODE REVIEW

File: Pod/Classes/UIControlBlocks/UIControl+Blocks.swift [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+extension UIControl: BlockControl {}
+
+public extension BlockControl where Self: UIControl {
+    /**
+     Set a block handler for an event.
+
+     - parameter event: A UIControlEvents event
+     - parameter callback: The event handler function
+     */
+    func setBlockTarget(for event: UIControlEvents, callback: @escaping (Self) -> Void) {
```
Comment (heyltsjay) [inline Pod/Classes/UIControlBlocks/UIControl+Blocks.swift]: #nth default to `.primaryActionTriggered`

Comment (heyltsjay) [inline Pod/Classes/UIControlBlocks/UIControl+Blocks.swift]: #pp this notation seems outdated. RE: Closures vs. blocks. how about `setCallback(for _, callback: _)`

**Comment by heyltsjay** [comment]
@KingOfBrian This may be nice for 99% of use cases:

```
    var callback: ((Self) -> Void)? {
        set {
            if let newValue = newValue {
                setBlockTarget(for: .primaryActionTriggered, callback: newValue)
            }
            else {
                clearBlockTargets(for: [.primaryActionTriggered])
            }
        }
        get {
            return blockTargets[UIControlEvents.primaryActionTriggered.rawValue]?.callback
        }
    }
```

**Comment by KingOfBrian** [comment]
Yea, I like it. Maybe `primaryCallback`?

File: Pod/Classes/UIControlBlocks/UIControl+Blocks.swift (hunk 1 above)
Comment (KingOfBrian) [reply Pod/Classes/UIControlBlocks/UIControl+Blocks.swift]: `setCallback(for event: UIControlEvent, _ callback: ...)` maybe? I don't like the double `callback` in the signature.

Comment (heyltsjay) [reply Pod/Classes/UIControlBlocks/UIControl+Blocks.swift]: that looks good

**Comment by heyltsjay** [comment]
@KingOfBrian Although, as currently written, this mutates the object. Also `.primaryActionTriggered` seems to be unimplemented for for vanilla `UIControl`s

**Comment by heyltsjay** [comment]
After a little while of use, I'm finding that I hardly ever use the control reference that is passed to the closure. I would exclude it, and have the minority use case reference via closure capture.

**Comment by octohub** [comment]
@KingOfBrian @heyltsjay Spring cleaning, let's merge or close this PR.

**Comment by ZevEisenberg** [comment]
@KingOfBrian @heyltsjay this was auto-closed due to some branching shenanigans I'm doing on Swiftilites. Please reopen if you want it to exist, or else please delete the branch.

**Pull request closed without merging**